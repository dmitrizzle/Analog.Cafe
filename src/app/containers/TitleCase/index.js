// tools
import React from "react"
import toTitleCase from "titlecase"

// components
import {
  TitleTextarea,
  SubtitleTextarea
} from "../../components/_controls/InputStyles"

// return
const components = {
  title: TitleTextarea,
  subtitle: SubtitleTextarea
}

export default class extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = { value: props.value || "" }
    this.handleChange = this.handleChange.bind(this)
  }
  handleChange = event => {
    this.setState({ value: toTitleCase(event.target.value) })
    this.props.onChange(event.target.value)
  }
  render = () => {
    const InputElement = components[this.props.inputDesignation]
    return (
      <InputElement
        autoFocus={this.props.autoFocus}
        value={this.state.value}
        onChange={this.handleChange}
        className={this.props.className}
        placeholder={this.props.placeholder}
        warning={this.props.warning}
        caution={this.props.caution}
        maxLength={this.props.maxLength}
        onKeyPress={this.props.onKeyPress}
      />
    )
  }
}
