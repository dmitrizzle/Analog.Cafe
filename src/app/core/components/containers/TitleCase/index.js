// tools
import React from "react"
import toTitleCase from "titlecase"

import { INPUT_AUTO_FORMAT } from "../../../../user/constants/input"
import {
  SubtitleTextarea,
  TitleTextarea
} from "../../stateless/_controls/InputStyles"

// return
const components = {
  title: TitleTextarea,
  subtitle: SubtitleTextarea
}

export default class extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = { value: props.value || "" }
    this.handleChange = this.handleChange.bind(this)
  }
  handleChange = event => {
    // preserve caret position
    const caret = event.target.selectionEnd
    const element = event.target
    window.requestAnimationFrame(() => {
      element.selectionStart = caret
      element.selectionEnd = caret
    })
    this.setState({
      value: INPUT_AUTO_FORMAT(toTitleCase(event.target.value))
    })
    this.props.onChange(event.target.value)
  }
  render = () => {
    const InputElement = components[this.props.inputDesignation]
    return (
      <InputElement
        ref="input"
        autoFocus={this.props.autoFocus}
        value={this.state.value}
        onChange={this.handleChange}
        className={this.props.className}
        placeholder={this.props.placeholder}
        warning={this.props.warning}
        caution={this.props.caution}
        maxLength={this.props.maxLength}
        onKeyPress={this.props.onKeyPress}
      />
    )
  }
}
