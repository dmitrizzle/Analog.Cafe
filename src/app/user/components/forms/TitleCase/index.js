import React from "react"
import toTitleCase from "titlecase"

import { INPUT_FORMAT } from "../../../constants/rules-submission"
import SubtitleTextarea from "../TextInput/components/SubtitleTextarea"
import TitleTextarea from "../TextInput/components/TitleTextarea"

const components = {
  title: TitleTextarea,
  subtitle: SubtitleTextarea
}
export default class extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = { value: props.value || "" }
    this.handleChange = this.handleChange.bind(this)
  }
  handleChange = event => {
    const caret = event.target.selectionEnd
    const element = event.target
    window.requestAnimationFrame(() => {
      element.selectionStart = caret
      element.selectionEnd = caret
    })
    this.setState({
      value: INPUT_FORMAT(toTitleCase(event.target.value))
    })
    this.props.onChange(event.target.value)
  }
  render = () => {
    const InputElement = components[this.props.inputDesignation]
    return (
      <InputElement
        ref="input"
        autoFocus={this.props.autoFocus}
        value={this.state.value}
        onChange={this.handleChange}
        className={this.props.className}
        placeholder={this.props.placeholder}
        warning={this.props.warning}
        caution={this.props.caution}
        maxLength={this.props.maxLength}
        onKeyPress={this.props.onKeyPress}
      />
    )
  }
}
