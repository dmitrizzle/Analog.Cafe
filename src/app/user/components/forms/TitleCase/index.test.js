import { shallow } from "enzyme"
import React from "react"

import TitleCase from "./"

it("Render TitleCase without crashing", () => {
  shallow(<TitleCase inputDesignation="title" />)
  shallow(<TitleCase inputDesignation="subtitle" />)
})
