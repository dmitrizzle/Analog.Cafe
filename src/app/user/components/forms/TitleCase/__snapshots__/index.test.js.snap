// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TitleCase, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent
    inputDesignation="title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": undefined,
      "caution": undefined,
      "className": undefined,
      "maxLength": undefined,
      "onChange": [Function],
      "onKeyPress": undefined,
      "placeholder": undefined,
      "value": "",
      "warning": undefined,
    },
    "ref": "input",
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": undefined,
        "caution": undefined,
        "className": undefined,
        "maxLength": undefined,
        "onChange": [Function],
        "onKeyPress": undefined,
        "placeholder": undefined,
        "value": "",
        "warning": undefined,
      },
      "ref": "input",
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render TitleCase, matches snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent
    inputDesignation="subtitle"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": undefined,
      "caution": undefined,
      "className": undefined,
      "maxLength": undefined,
      "onChange": [Function],
      "onKeyPress": undefined,
      "placeholder": undefined,
      "value": "",
      "warning": undefined,
    },
    "ref": "input",
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": undefined,
        "caution": undefined,
        "className": undefined,
        "maxLength": undefined,
        "onChange": [Function],
        "onKeyPress": undefined,
        "placeholder": undefined,
        "value": "",
        "warning": undefined,
      },
      "ref": "input",
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
